plugin.video.youtube/resources/lib/youtube_plugin/youtube/client/youtube.py

```diff
@@ -655,13 +674,29 @@ def _perform(_page_token, _offset, _result, _shelf_index=None):
                 _items = _items[_offset:]
             _result['offset'] = _new_offset
 
-            for _item in _items:
-                _item = _item.get('gridVideoRenderer', {})
+            for _listItem in _items:
+                _item = _listItem.get('gridVideoRenderer', {})
                 if _item:
                     _video_item = {'id': _item['videoId'],
                                    'title': _item.get('title', {}).get('runs', [{}])[0].get('text', ''),
                                    'channel': _item.get('shortBylineText', {}).get('runs', [{}])[0].get('text', '')}
                     _result['items'].append(_video_item)
+                _item = _listItem.get('gridPlaylistRenderer', {})
+                if _item:
+                    play_next_page_token = ''
+                    while True:
```

Comment:
if you need the "infinite loop" maybe you should check for abort_requested in an if condition below

Reply:
added a check for abort_requested